packages/react-reconciler/src/ReactFiberHooks.js

```diff
@@ -753,6 +754,33 @@ function renderWithHooksAgain<Props, SecondArg>(
   return children;
 }
 
+export function renderStatefulHostComponentWithHooks(
+  current: Fiber | null,
+  workInProgress: Fiber,
+  lanes: Lanes,
+): boolean {
+  if (!enableFormActions) {
+    return false;
+  }
+  return renderWithHooks(
```

Comment:
This overrides `updateQueue` to null. That field is fine to reuse after `diffInCommitPhase` and outside of legacy mode hydration I think.

But just setting it to null here is probably also fine.

Reply:
I wonder what the specialized form of this looks like for this use case when we can ignore all the user space stuff like multiple render passes, DEV warnings. We don't even have to set the ReactCurrentDispatcher.

Reply:
Yeah I explored during that in this diff but it started getting out of hand so I punted on that for now. Maybe once there's 2 cases.